pkg/deploy/registry/instantiate/rest.go

```diff
@@ -0,0 +1,286 @@
+package instantiate
+
+import (
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/rest"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/registry/generic/registry"
+	"k8s.io/kubernetes/pkg/runtime"
+	utilerrors "k8s.io/kubernetes/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	"github.com/openshift/origin/pkg/client"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/deploy/api/validation"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+func NewREST(store registry.Store, oc client.Interface, kc kclient.Interface, codec runtime.Codec) *REST {
+	store.UpdateStrategy = Strategy
+	return &REST{store: &store, isn: oc, rn: kc, codec: codec}
+}
+
+// REST implements the Updater interface.
+var _ = rest.Updater(&REST{})
+
+type REST struct {
+	store *registry.Store
+	isn   client.ImageStreamsNamespacer
+	rn    kclient.ReplicationControllersNamespacer
+	codec runtime.Codec
+}
+
+func (s *REST) New() runtime.Object {
+	return &deployapi.DeploymentRequest{}
+}
+
+// Update instantiates a deployment config
+func (r *REST) Update(ctx kapi.Context, name string, objInfo rest.UpdatedObjectInfo) (runtime.Object, bool, error) {
+	configObj, err := r.store.Get(ctx, name)
+	if err != nil {
+		return nil, false, err
+	}
+	config := configObj.(*deployapi.DeploymentConfig)
+
+	old := deployapi.RequestForConfig(config)
+	obj, err := objInfo.UpdatedObject(ctx, old)
+	if err != nil {
+		return nil, false, err
+	}
+
+	req, ok := obj.(*deployapi.DeploymentRequest)
+	if !ok {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("wrong object passed to DeploymentRequest update: %#v", obj))
+	}
+
+	if errs := validation.ValidateDeploymentRequest(req); len(errs) > 0 {
+		return nil, false, errors.NewInvalid(deployapi.Kind("DeploymentRequest"), req.Name, errs)
+	}
+
+	// TODO: Consider instantiating paused deployments.
```

Comment:
The thing is that we don't have this upstream currently and I am hesitant in doing this for us before having it upstream. Can't we enable it later?